modules/avatar/react/lib/Avatar.tsx

```diff
@@ -72,14 +76,29 @@ export default class Avatar extends React.Component<AvatarProps> {
     altText: 'Avatar',
   };
 
+  constructor(props: AvatarProps) {
+    super(props);
+    this.state = {
+      hasImageError: false,
+    };
+  }
+
+  public setFallbackIcon = () => {
+    if (!this.state.hasImageError) {
+      this.setState({
+        hasImageError: true,
+      });
+    }
+  };
+
   render() {
     const {variant, altText, size, url, ...elemProps} = this.props;
 
     const background = variant === AvatarVariant.Dark ? colors.blueberry400 : colors.soap300;
     return (
       <StyledContainer variant={variant} size={size} aria-label={altText} {...elemProps}>
-        {url ? (
-          <img src={url} alt={altText} />
+        {url && !this.state.hasImageError ? (
+          <img src={url} alt={altText} onError={this.setFallbackIcon} />
         ) : (
           <SystemIconCircle icon={userIcon} background={background} size={size} />
         )}
```

Comment:
Is it possible to fade in a loaded avatar when loaded and not show on error instead of DOM swapping? If there is a URL, is the area simply blank until the URL is loaded?

Reply:
Oh that is a good idea. Let me play around with it

Reply:
Added a fade in over the top of the default icon, lmkwyt